server.js

```diff
@@ -1,17 +1,78 @@
 // import modules
-const bodyParser = require('body-parser');
-const cors = require('cors');
+// const cors = require('cors'); // for react app use later
 const express = require('express');
 const helmet = require('helmet');
 const morgan = require('morgan');
 
+// import helpers
+const userDb = require('./data/helpers/userDb.js');
+
 // init server
 const server = express();
 
 // apply middleware
-server.use(bodyParser());
-server.use(cors());
+server.use(express.json());
 server.use(helmet());
 server.use(morgan('short'));
 
+// fetch users
+server.get('/api/users', async (req, res) => {
+    try {
+        const users = await userDb.get();
+        res.status(200).json(users);
+    } catch (err) {
+        res.status(500).json({
+            message: 'There was an error while retrieving users.',
+            error: err
+        });
+    }
+});
+
+// fetch user by ID
+server.get('/api/users/:id', async (req, res) => {
+    const { id } = req.params;
+    try {
+        const user = await userDb.get(id);
+        if (user) {
+            res.status(200).json(user)
+        } else {
+            res.status(404).json({
+                message: 'User not found.'
+            })
+        }
+    } catch(err) {
+        res.status(500).json({
+            message: 'There was an error while retrieving the user.',
+            error: err
+        });
+    }
+});
+
+// add new user
+server.post('/api/users', async (req, res) => {
+    const name = req.body;
+
+    function upperCase(name) {
+        return name.charAt(0).toUpperCase() + name.slice(1);
+    }
+
+    if (!name || name.length > 128) {
+        res.status(400).json({
+            message: 'Invalid username'
+        });
+    } else {
+        name.name = upperCase(name.name);
```

Comment:
This seems a little repetitive, but I assume this is done to maintain access to the object. I'd personally prefer this: `const { body } = req` and then, `body.name = upperCase(body.name)`, because it seems a little less confusing.